src/plugins/discover/public/application/apps/main/discover_main_route.tsx

```diff
@@ -86,6 +74,9 @@ export function DiscoverMainRoute(props: DiscoverMainProps) {
 
     async function loadSavedSearch() {
       try {
+        // force a refresh if a given saved search without id was saved
+        setSavedSearch(undefined);
+
```

Comment:
ok, don't like this but it forces the trigger a request when we save a new saved search. then savedSearch searchSource is updated correctly before this request

Reply:
sure this can be solve in a more elegant way